/***************************************************************************
 *  WorkpieceAddition.proto - LLSF Protocol - Addition to put in Worpieces
 *
 *  Created: Mon Feb 11 16:35:46 2013
 *  Copyright  2015  Randolph Maa√üen
 ****************************************************************************/

/*  Redistribution and use in source and binary forms, with or without
 *  modification, are permitted provided that the following conditions
 *  are met:
 *
 * - Redistributions of source code must retain the above copyright
 *   notice, this list of conditions and the following disclaimer.
 * - Redistributions in binary form must reproduce the above copyright
 *   notice, this list of conditions and the following disclaimer in
 *   the documentation and/or other materials provided with the
 *   distribution.
 * - Neither the name of the authors nor the names of its contributors
 *   may be used to endorse or promote products derived from this
 *   software without specific prior written permission.
 *
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
 * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
 * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
 * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
 * COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
 * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
 * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
 * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
 * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
 * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
 * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
 * OF THE POSSIBILITY OF SUCH DAMAGE.
 */

package gazsim_msgs;

// The possible colors of the ring
enum Color{
  RED = 0;
  GREEN = 1;
  BLUE = 2;
  GREY = 3;
  BLACK = 4;
  YELLOW = 5;
  ORANGE = 6;
  SILVER = 7;
  NONE = 8;
}

enum Command{
  ADD_RING = 0;
  ADD_CAP = 1;
  REMOVE_CAP = 2;
  DELIVER = 3;
}

enum Team {
  CYAN = 0;
  MAGENTA = 1;
}

// Time stamp and duration structure.
// Can be used for absolute times or
// durations alike.
message WorkpieceCommand {

  required Command command = 1;
  repeated Color color = 2;
  required string puck_name = 3;
  optional Team team_color = 4;

}

message WorkpieceResult {
  required Color color = 1;
  required string puck_name = 2;
}
